src/boosting/gbdt.cpp

```diff
@@ -16,9 +16,16 @@
 
 namespace LightGBM {
 
-GBDT::GBDT() 
+GBDT::GBDT()
   :iter_(0),
+  train_data_(nullptr),
```

Comment:
If we don't initialize `train_data_`, `EXC_BAD_ACCESS` occurs at https://github.com/Microsoft/LightGBM/blob/master/src/boosting/gbdt.cpp#L448 because of https://github.com/Microsoft/LightGBM/blob/master/src/boosting/gbdt.cpp#L443-L445